sdk/keyvault/azcertificates/constants.go

```diff
@@ -0,0 +1,113 @@
+	// available. System wil permanently delete it after 90 days, if not recovered
+	DeletionRecoveryLevelRecoverable DeletionRecoveryLevel = "Recoverable"
+	// DeletionRecoveryLevelRecoverableProtectedSubscription - Denotes a vault and subscription state in which deletion is recoverable within retention interval
+	// (90 days), immediate and permanent deletion (i.e. purge) is not permitted, and in which the subscription itself cannot be permanently canceled. System
+	// wil permanently delete it after 90 days, if not recovered
+	DeletionRecoveryLevelRecoverableProtectedSubscription DeletionRecoveryLevel = "Recoverable+ProtectedSubscription"
+	// DeletionRecoveryLevelRecoverablePurgeable - Denotes a vault state in which deletion is recoverable, and which also permits immediate and permanent deletion
+	// (i.e. purge). This level guarantees the recoverability of the deleted entity during the retention interval (90 days), unless a Purge operation is requested,
+	// or the subscription is cancelled. System wil permanently delete it after 90 days, if not recovered
+	DeletionRecoveryLevelRecoverablePurgeable DeletionRecoveryLevel = "Recoverable+Purgeable"
+)
+
+// ToPtr returns a *DeletionRecoveryLevel pointing to the current value.
+func (c DeletionRecoveryLevel) ToPtr() *DeletionRecoveryLevel {
+	return &c
+}
+
+// JSONWebKeyCurveName - Elliptic curve name. For valid values, see JsonWebKeyCurveName.
+type JSONWebKeyCurveName string
+
+const (
+	JSONWebKeyCurveNameP256  JSONWebKeyCurveName = "P-256"
+	JSONWebKeyCurveNameP256K JSONWebKeyCurveName = "P-256K"
+	JSONWebKeyCurveNameP384  JSONWebKeyCurveName = "P-384"
+	JSONWebKeyCurveNameP521  JSONWebKeyCurveName = "P-521"
+)
+
+// ToPtr returns a *JSONWebKeyCurveName pointing to the current value.
+func (c JSONWebKeyCurveName) ToPtr() *JSONWebKeyCurveName {
+	return &c
+}
+
+// JSONWebKeyType - The type of key pair to be used for the certificate.
+type JSONWebKeyType string
+
+const (
+	JSONWebKeyTypeEC     JSONWebKeyType = "EC"
+	JSONWebKeyTypeECHSM  JSONWebKeyType = "EC-HSM"
+	JSONWebKeyTypeOct    JSONWebKeyType = "oct"
+	JSONWebKeyTypeOctHSM JSONWebKeyType = "oct-HSM"
+	JSONWebKeyTypeRSA    JSONWebKeyType = "RSA"
+	JSONWebKeyTypeRSAHSM JSONWebKeyType = "RSA-HSM"
+)
+
+// ToPtr returns a *JSONWebKeyType pointing to the current value.
+func (c JSONWebKeyType) ToPtr() *JSONWebKeyType {
+	return &c
+}
+
+type KeyUsageType string
+
+const (
+	KeyUsageTypeCRLSign          KeyUsageType = "cRLSign"
+	KeyUsageTypeDataEncipherment KeyUsageType = "dataEncipherment"
+	KeyUsageTypeDecipherOnly     KeyUsageType = "decipherOnly"
+	KeyUsageTypeDigitalSignature KeyUsageType = "digitalSignature"
+	KeyUsageTypeEncipherOnly     KeyUsageType = "encipherOnly"
+	KeyUsageTypeKeyAgreement     KeyUsageType = "keyAgreement"
+	KeyUsageTypeKeyCertSign      KeyUsageType = "keyCertSign"
+	KeyUsageTypeKeyEncipherment  KeyUsageType = "keyEncipherment"
+	KeyUsageTypeNonRepudiation   KeyUsageType = "nonRepudiation"
+)
```

Comment:
`CertificateKeyUsage`: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateKeyUsage.cs